miner/worker.go

```diff
@@ -829,12 +829,20 @@ func (w *worker) applyTransaction(env *environment, tx *types.Transaction, recei
 		snap = env.state.Snapshot()
 		gp   = env.gasPool.Gas()
 	)
+	gasPrice, err := tx.EffectiveGasTip(env.header.BaseFee)
+	if err != nil {
+		return nil, err
+	}
 
 	receipt, err := core.ApplyTransaction(env.evm, env.gasPool, env.state, env.header, tx, &env.header.GasUsed, receiptProcessors...)
 	if err != nil {
 		env.state.RevertToSnapshot(snap)
 		env.gasPool.SetGas(gp)
 	}
+
+	gasUsed := new(big.Int).SetUint64(receipt.GasUsed)
+	env.profit.Add(env.profit, gasUsed.Mul(gasUsed, gasPrice))
```

Comment:
Could you explain why we need to add profit here? Was there a bug before this commit?

Reply:
The original calculation method only accounted for the profit of bundles. I think including the profit of regular transactions will make the bid more accurate. Flashbots' builder has also implemented this.